node/pkg/fetcher/app_test.go

```diff
@@ -116,3 +118,76 @@ func TestAppRun(t *testing.T) {
 	}
 	assert.Greater(t, len(localAggregateResult), 0)
 }
+
+func TestLocalAggregatesChannel(t *testing.T) {
+	ctx := context.Background()
+	clean, testItems, err := setup(ctx)
+	if err != nil {
+		t.Fatalf("error setting up test: %v", err)
+	}
+	defer func() {
+		if cleanupErr := clean(); cleanupErr != nil {
+			t.Logf("Cleanup failed: %v", cleanupErr)
+		}
+	}()
+	app := testItems.app
+
+	// get configs, initialize channel, and start collectors
+	configs, err := app.getConfigs(ctx)
+	if err != nil {
+		t.Fatalf("error getting configs: %v", err)
+	}
+
+	localAggregatesChannel := make(chan LocalAggregate, LocalAggregatesChannelSize)
+	app.Collectors = make(map[int32]*Collector, len(configs))
+
+	feedData := make(map[string]any)
+	for _, config := range configs {
+		collectorFeeds, getFeedsErr := app.getFeeds(ctx, config.ID)
+		if getFeedsErr != nil {
+			t.Fatalf("error getting configs: %v", getFeedsErr)
+		}
+		app.Collectors[config.ID] = NewCollector(config, collectorFeeds, localAggregatesChannel)
+		for _, feed := range collectorFeeds {
+			feedData[keys.LatestFeedDataKey(feed.ID)] = FeedData{FeedID: feed.ID, Value: DUMMY_FEED_VALUE, Timestamp: nil, Volume: DUMMY_FEED_VALUE}
+		}
+	}
+	err = app.startAllCollectors(ctx)
+	if err != nil {
+		t.Fatalf("error starting collectors: %v", err)
+	}
+
+	err = db.MSetObject(ctx, feedData)
+	if err != nil {
+		t.Fatalf("error setting feed data in redis: %v", err)
+	}
+
+	// wait until collectors fetch and process data from db
+	localAggregateIntervalRaw := os.Getenv("LOCAL_AGGREGATE_INTERVAL")
+	localAggregateInterval, err := time.ParseDuration(localAggregateIntervalRaw)
+	if err != nil {
+		localAggregateInterval = DefaultLocalAggregateInterval
+	}
+	time.Sleep(localAggregateInterval*2)
```

Comment:
have you tried running test without sleeps? if no error occurs without sleeps, lets remove all sleeps inside

Reply:
apparently the first sleep was not necessary but the second is required because it takes a bit longer to insert data to db